datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnectorTask.java

```diff
@@ -241,6 +244,16 @@ protected DatastreamProducerRecord translate(ConsumerRecord<?, ?> fromKafka, Ins
     String offsetStr = String.valueOf(offset);
     metadata.put(KAFKA_ORIGIN_OFFSET, offsetStr);
     metadata.put(BrooklinEnvelopeMetadataConstants.EVENT_TIMESTAMP, String.valueOf(eventsSourceTimestamp));
+    if (Boolean.TRUE.toString()
+        .equals(_datastream.getMetadata().get(KafkaDatastreamMetadataConstants.USE_PASSTHROUGH_COMPRESSION))) {
+      // If passthrough mode is enabled, we need to create a Kafka header on the transport side for supporting
+      // Kafka broker message format bump. The magic byte contains details about the message format for the passthrough
+      // record and it needs to be preserved and set via the Kafka headers to ensure that the correct message format
+      // can be negotiated.
+      PassThroughConsumerRecord<?, ?> passThroughConsumerRecord = (PassThroughConsumerRecord<?, ?>) fromKafka;
+      metadata.put(KafkaPassthroughRecordMagicConverter.PASS_THROUGH_MAGIC_VALUE,
```

Comment:
This patch LGTM. Any reason, you want to convert this to a String and not use it as it is?

Reply:
@MayureshGharat we convert ConsumerRecord to a `DatastreamProducerRecord`, and `DatastreamProdcuerRecord` has a list  of `BrooklinEnvelope`. `BrooklinEnvelope` has a generic field which stores all "metadata" and is a `Map<String, String>` . If I didn't convert the magic to String to store in this generic "metadata", I'd have to introduce a new field to store the magic as a `byte`. Which isn't a bad option, but this will only be used by BMM passthrough. 

@DEEPTHIKORAT and @jzakaryan can you folks also comment on whether using "metadata" and doing string conversion makes more sense, or it's okay to introduce this `byte` field to store the magic? The only use case would be BMM passthrough.

Reply:
Here's my two cents. @somandal I wouldn't introduce a "magic" byte as a separate field, because it only makes sense in the scope of BMM passthrough, and it would be useless and confusing out of that limited scope.
The only precaution with the current approach would be the conversion overhead, but since it's just being done once for a batch, it should be fine.

Reply:
@jzakaryan thank you!